library/c/api.go

```diff
@@ -92,25 +88,25 @@ func main() {}
 // - dns4DomainName: the domain name resolving to the node's public IPv4 address.
 //
 //export waku_new
-func waku_new(configJSON *C.char, onErrCb C.WakuCallBack) C.int {
+func waku_new(configJSON *C.char, cb C.WakuCallBack, userData unsafe.Pointer) C.int {
 	err := library.NewNode(C.GoString(configJSON))
-	return execErrCB(onErrCb, err)
+	return execErrCB(err, cb, userData)
```

Comment:
This name is confusing, the function actually executes callback and internally handles error, maybe we should rename it to something else.